### PR TITLE
tests: introduce catch2 unit tests infrastructure.

### PR DESCRIPTION
**Pull request by hermet**
Catch2 is a multi-paradigm test framework for C++.
It is primarily distributed as a single header file,
very easy and simple to adopt this to thorvg project.

This patch introduces catch2 infrastsructure and one prototype as a sample.

You can refer "testInitializer.cpp", how to add unit test! (nothing else)

Also, enable Unit-tests with meson option when you change any thorvg code.

$meson build -Dtests=true.

launch tvgUnitTest in the build result then verify 100% coverage
before submitting any patches.

For your information, you can check test option by this command:
$tvgUnitTest -h

### CODE REVIEW

File: test/meson.build [hunk 1]
```diff
@@ -0,0 +1,9 @@
+test_file = [
+    'testMain.cpp',
+    'testInitializer.cpp',
+]
+
+executable('tvgUnitTests',
+           test_file,
+           include_directories : headers,
+           link_with : thorvg_lib)
```
Comment (JSUYA) [inline test/meson.build:9]: If we use test() of meson script, we can run test with ninja command.
ex) `ninja -C build test` command.
```
unitTests = executable('tvgUnitTests',
           test_file,
           include_directories : headers,
           link_with : thorvg_lib)

test('Unit test', unitTests)

```

File: test/testInitializer.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+   REQUIRE(Initializer::term(CanvasEngine::Sw) == Result::Success);
+   REQUIRE(Initializer::term(CanvasEngine::Sw) == Result::Success);
+}
+
+TEST_CASE("Negative termination", "[tvgInitializer]")
+{
+   REQUIRE(Initializer::term(CanvasEngine::Sw) == Result::InsufficientCondition);
+}
+
+TEST_CASE("Many threads", "[tvgInitializer]")
```
Comment (JSUYA) [inline test/testInitializer.cpp:51]: Many? is this typo?

File: test/testInitializer.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+{
+   REQUIRE(Initializer::init(CanvasEngine::Sw, -1) == Result::FailedAllocation);
```
Comment (JSUYA) [inline test/testInitializer.cpp:53]: This unit test will fail. Is it intentional?
```
-------------------------------------------------------------------------------
Many threads
-------------------------------------------------------------------------------
../test/testInitializer.cpp:51
...............................................................................

../test/testInitializer.cpp:53: FAILED:
  REQUIRE( Initializer::init(CanvasEngine::Sw, -1) == Result::FailedAllocation )
with expansion:
  2 == 3

===============================================================================
test cases:  5 |  4 passed | 1 failed
assertions: 11 | 10 passed | 1 failed

```

**Review by JSUYA — COMMENTED** [review]
Welcome to catch2 :)

**Comment by JSUYA** [comment]
> **CODING STYLE CHECK**:
> 
> ```
> testInitializer.cpp:31:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:32:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:37:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:38:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:39:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:41:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:42:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:43:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:48:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:53:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:54:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> testInitializer.cpp:59:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> Done processing testInitializer.cpp
> Done processing testMain.cpp
> Total errors found: 12
> ```

please take a look here ~
We are 4 spaces indentation

File: test/testInitializer.cpp (hunk 3 above)
Comment (hermet) [reply test/testInitializer.cpp:53]: @JSUYA no, i must be passed.

File: test/testInitializer.cpp (hunk 2 above)
Comment (hermet) [reply test/testInitializer.cpp:51]: @JSUYA   "Many Threads" is intended like  uint32 threadsCnt = -1;

**Comment by hermet** [comment]
> > **CODING STYLE CHECK**:
> > ```
> > testInitializer.cpp:31:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:32:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:37:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:38:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:39:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:41:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:42:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:43:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:48:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:53:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:54:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > testInitializer.cpp:59:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
> > Done processing testInitializer.cpp
> > Done processing testMain.cpp
> > Total errors found: 12
> > ```
> 
> please take a look here ~
> We are 4 spaces indentation

fixed!

**Review by JSUYA — APPROVED** [review]
LGTM

**Pull request merged**